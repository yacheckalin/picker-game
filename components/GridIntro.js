import React, { useEffect } from "react";

const GridIntro = React.memo((props) => {
  useEffect(() => {
    const callback = function () {
      const elems = document.querySelectorAll(".collapsible");
      M.Collapsible.init(elems, {});
    };
    document.addEventListener("DOMContentLoaded", callback);
    return () => {
      document.removeEventListener("DOMContentLoaded", callback);
    };
  });
  return (
    <div className="col s12 card-panel lime lighten-4">
      <blockquote>
        <ul className="collapsible">
          <li className="active">
            <div className="collapsible-header">
              <i className="material-icons">beach_access</i>The Mission:
            </div>
            <div className="collapsible-body">
              The mission is to find the exit from the room!
            </div>
          </li>
          <li>
            <div className="collapsible-header">
              <i className="material-icons">dashboard</i>Navigation
            </div>
            <div className="collapsible-body">
              <p>
                <i className="material-icons">arrow_back</i> - go left
              </p>
              <p>
                <i className="material-icons">arrow_forward</i> - go right
              </p>
              <p>
                <i className="material-icons">arrow_upward</i> - go up
              </p>

              <p>
                <i className="material-icons">arrow_downward</i> - go down
              </p>
              <p>
                <i className="material-icons">photo_camera</i> [Alt] - observe
              </p>
              <p>
                <i className="material-icons">publish</i> [Space] - pick an
                object
              </p>
              <p>
                <i className="material-icons">lock_open</i> [Enter] - open the
                door
              </p>
            </div>
          </li>
        </ul>
      </blockquote>
    </div>
  );
});

export default GridIntro;
