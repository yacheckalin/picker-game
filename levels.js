import { KEYS, DOORS, WALL, VISITED } from "./constants";

export const LEVELS = [
  [
    "level_1",
    [
      [VISITED, 0, WALL, WALL, 1, 1, 1, 1],
      [1, 0, 1, 0, 0, 0, KEYS.RED_KEY, 1],
      [1, 0, 1, 0, 1, 1, 1, 1],
      [1, 0, 1, 0, 1, 0, KEYS.BLUE_KEY, 1],
      [1, 0, 0, 0, 1, 0, 0, 1],
      [1, 0, 1, 0, 1, 0, 0, 1],
      [1, KEYS.GREEN_KEY, 1, 0, 0, 0, 0, DOORS.BLUE_DOOR],
      [1, 1, 1, 1, 1, 1, 1, 1],
    ],
  ],
  [
    "level_2",
    [
      [VISITED, 0, WALL, WALL, 1, 1, 1, 1],
      [1, 0, 1, 0, 0, 0, 0, 1],
      [1, 0, 1, 0, 1, 1, 1, 1],
      [1, 0, 1, 0, 1, 0, KEYS.BLUE_KEY, 1],
      [1, 0, 0, 0, 1, 0, 0, 1],
      [1, 0, 1, 0, 1, 0, 0, 1],
      [1, 0, 1, 0, 0, 0, 0, DOORS.BLUE_DOOR],
      [1, 1, 1, 1, 1, 1, 1, 1],
    ],
  ],
];
