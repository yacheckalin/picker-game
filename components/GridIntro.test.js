import React from "react";
import GridIntro from "./GridIntro";
import toJSON from "enzyme-to-json";
import { shallow, mount } from "enzyme";

import { LEVELS } from "../levels";

describe("<GridIntro />", () => {
  const defaultProps = {
    loadHandler: jest.fn(),
  };

  test("render properly", () => {
    const wrapper = shallow(<GridIntro {...defaultProps} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("render collapsible list on a page", () => {
    const wrapper = shallow(<GridIntro {...defaultProps} />);

    expect(wrapper.find(`collapsible`)).toBeTruthy();
  });

  test("render entire list of levels properly", () => {
    const wrapper = shallow(<GridIntro {...defaultProps} />);
    for (const [hash, grid, size, mission] of LEVELS) {
      expect(wrapper.find(hash)).toBeTruthy();
    }
  });

  test("when select level from the list, mission changes", () => {
    const wrapper = shallow(<GridIntro {...defaultProps} />);
    wrapper.find("select").simulate("change", { target: { value: "1" } });

    expect(wrapper.text()).toContain(`${LEVELS[1][3]}`);
  });

  test("render navigation explanations", () => {
    const wrapper = shallow(<GridIntro {...defaultProps} />);
    expect(wrapper.find("open the door")).toBeTruthy();
    expect(wrapper.find("pick")).toBeTruthy();
    expect(wrapper.find("go up")).toBeTruthy();
    expect(wrapper.find("go down")).toBeTruthy();
    expect(wrapper.find("go left")).toBeTruthy();
    expect(wrapper.find("go right")).toBeTruthy();
  });
});
