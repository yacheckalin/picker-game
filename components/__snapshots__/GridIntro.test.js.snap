// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridIntro /> render properly 1`] = `
<div
  className="col s12 card-panel lime lighten-4"
>
  <div
    className="row"
  />
  <blockquote>
    <ul
      className="collapsible"
    >
      <li>
        <div
          className="collapsible-header"
        >
          <i
            className="material-icons"
          >
            flash_on
          </i>
           Load Level
        </div>
        <div
          className="collapsible-body"
        >
          <div
            className="input-field "
          >
            <select
              onChange={[Function]}
            >
              <option
                disabled={true}
                value=""
              >
                Choose your level
              </option>
              <option
                key="0"
                value={0}
              >
                level_zero
              </option>
              <option
                key="1"
                value={1}
              >
                level_1
              </option>
              <option
                key="2"
                value={2}
              >
                level_2
              </option>
              <option
                key="3"
                value={3}
              >
                level_3
              </option>
            </select>
            <label>
              Level
            </label>
          </div>
        </div>
      </li>
      <li
        className="active"
      >
        <div
          className="collapsible-header"
        >
          <i
            className="material-icons"
          >
            beach_access
          </i>
          The Mission:
        </div>
        <div
          className="collapsible-body"
        >
          Open the BLUE DOOR!
        </div>
      </li>
      <li>
        <div
          className="collapsible-header"
        >
          <i
            className="material-icons"
          >
            dashboard
          </i>
          Navigation
        </div>
        <div
          className="collapsible-body"
        >
          <p>
            <i
              className="material-icons"
            >
              arrow_back
            </i>
             - go left
          </p>
          <p>
            <i
              className="material-icons"
            >
              arrow_forward
            </i>
             - go right
          </p>
          <p>
            <i
              className="material-icons"
            >
              arrow_upward
            </i>
             - go up
          </p>
          <p>
            <i
              className="material-icons"
            >
              arrow_downward
            </i>
             - go down
          </p>
          <p>
            <i
              className="material-icons"
            >
              photo_camera
            </i>
             [Alt] - observe
          </p>
          <p>
            <i
              className="material-icons"
            >
              publish
            </i>
             [Space] - pick an object
          </p>
          <p>
            <i
              className="material-icons"
            >
              lock_open
            </i>
             [Enter] - open the door
          </p>
        </div>
      </li>
    </ul>
  </blockquote>
</div>
`;
